helix-core/src/test/java/org/apache/helix/integration/task/TestDeleteJobFromJobQueue.java

```diff
@@ -69,11 +70,16 @@ public void testForceDeleteJobFromJobQueue() throws InterruptedException {
     Assert
         .assertNotNull(_driver.getJobContext(TaskUtil.getNamespacedJobName(jobQueueName, "job2")));
 
-    // The following force delete for the job should go through without getting an exception
-    _driver.deleteJob(jobQueueName, "job2", true);
+    try {
```

Comment:
Another way can be to stop the controller and do force delete? Is there any reason you chose to use try-catch? I think stopping the controller and doing force delete can be a better approach as it actually checks the dele operation.

Reply:
Changed to pause pipeline with following code:

```
    // force deletion can have Exception thrown as controller is writing to propertystore path too.
    // https://github.com/apache/helix/issues/1406
    
    // Thus, we stop pipeline to make sure there is not such race condition.
    _gSetupTool.getClusterManagementTool().enableCluster(CLUSTER_NAME, false);
    
    _driver.deleteJob(jobQueueName, "job2", true);
    Thread.sleep(3000); // note this sleep is critical as it would take time for controller to stop

    // Check that the job has been force-deleted (fully gone from ZK)
    Assert.assertNull(_driver.getJobConfig(TaskUtil.getNamespacedJobName(jobQueueName, "job2")));
    Assert.assertNull(_driver.getJobContext(TaskUtil.getNamespacedJobName(jobQueueName, "job2")));


```